app/src/main/java/com/kickstarter/viewmodels/CreatorDashboardReferrerBreakdownHolderViewModel.java

```diff
@@ -0,0 +1,270 @@
+        .map(Float::intValue)
+        .compose(avg -> Observable.combineLatest(currentProject, avg, Pair::create));
+
+      this.customReferrerColor = Observable.just(ReferrerColor.CUSTOM.getReferrerColor())
+        .compose(Transformers.takeWhen(this.projectAndProjectStatsInput));
+
+      this.customReferrerPercent = customReferrers
+        .flatMap(rs ->
+          Observable.from(rs)
+            .reduce(0f, (accum, stat) -> accum + stat.percentageOfDollars())
+        );
+
+      this.customReferrerPercentText = this.customReferrerPercent
+        .map(percent -> NumberUtils.flooredPercentage(percent.floatValue() * 100f));
+
+      this.customReferrerPledgedAmount = customReferrers
+        .flatMap(rs ->
+          Observable.from(rs)
+            .reduce(0f, (accum, stat) -> accum + stat.pledged())
+        );
+
+      this.projectAndCustomReferrerPledgedAmount = Observable.combineLatest(
+        currentProject,
+        this.customReferrerPledgedAmount,
+        Pair::create
+      );
+
+      this.externalReferrerColor = Observable.just(ReferrerColor.EXTERNAL.getReferrerColor())
+        .compose(Transformers.takeWhen(this.projectAndProjectStatsInput));
+
+      this.externalReferrerPercent = externalReferrers
+        .flatMap(rs ->
+          Observable.from(rs)
+            .reduce(0f, (accum, stat) -> accum + stat.percentageOfDollars())
+        );
+
+      this.externalReferrerPercentText = this.externalReferrerPercent
+        .map(percent -> NumberUtils.flooredPercentage(percent * 100f));
+
+      this.externalReferrerPledgedAmount = customReferrers
+        .flatMap(rs ->
+          Observable.from(rs)
+            .reduce(0f, (accum, stat) -> accum + stat.pledged())
+        );
+
+      this.projectAndExternalReferrerPledgedAmount = Observable.combineLatest(
+        currentProject,
+        this.externalReferrerPledgedAmount,
+        Pair::create
+      );
+
+      this.internalReferrerColor = Observable.just(ReferrerColor.INTERNAL.getReferrerColor())
+        .compose(Transformers.takeWhen(this.projectAndProjectStatsInput));
+
+      this.internalReferrerPercent = internalReferrers
+        .flatMap(rs ->
+          Observable.from(rs)
+            .reduce(0f, (accum, stat) -> accum + stat.percentageOfDollars())
+        );
+
+      this.internalReferrerPercentText = this.internalReferrerPercent
+        .map(percent -> NumberUtils.flooredPercentage(percent.floatValue() * 100f));
+
+      this.internalReferrerPledgedAmount = internalReferrers
+        .flatMap(rs ->
+          Observable.from(rs)
+            .reduce(0f, (accum, stat) -> accum + stat.pledged())
+        );
+
+      this.projectAndInternalReferrerPledgedAmount = Observable.combineLatest(
+        currentProject,
+        this.internalReferrerPledgedAmount,
+        Pair::create
+      );
+
+      this.pledgedViaCustomLayoutIsGone = this.customReferrerPledgedAmount
+        .map(amount -> amount > 0f)
+        .map(BooleanUtils::negate);
```

Comment:
rather than having a `.map(BooleanUtils::negate);` here and L163, 167, it's probably just more straightforward to change the logic in the `amount` lambda to be something like `amount <= 0f`

Reply:
oh lol tru